internal/controller/config/config.go

```diff
@@ -0,0 +1,155 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/spf13/viper"
+
+	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
+)
+
+type ConfigManager struct {
+	config        *GlobalConfig
+	rolloutConfig *NumaflowControllerDefinitionConfig
+	lock          *sync.RWMutex
+}
+
+var instance *ConfigManager
+var once sync.Once
+
+// GetConfigManagerInstance  returns a singleton config manager throughout the application
+func GetConfigManagerInstance() *ConfigManager {
+	once.Do(func() {
+		instance = &ConfigManager{
+			config:        &GlobalConfig{},
+			rolloutConfig: &NumaflowControllerDefinitionConfig{},
+			lock:          new(sync.RWMutex),
+		}
+	})
+	return instance
+}
+
+// GlobalConfig is the configuration for the controllers, it is
+// supposed to be populated from the configmap attached to the
+// controller manager.
+type GlobalConfig struct {
+	ClusterName             string `json:"clusterName" mapstructure:"clusterName"`
+	SyncTimeIntervalMs      int    `json:"syncTimeIntervalMs" mapstructure:"syncTimeIntervalMs"`
+	AutomatedSyncDisabled   bool   `json:"automatedSyncDisabled" mapstructure:"automatedSyncDisabled"`
+	CascadeDeletion         bool   `json:"cascadeDeletion" mapstructure:"cascadeDeletion"`
+	AutoHealDisabled        bool   `json:"autoHealDisabled" mapstructure:"autoHealDisabled"`
```

Comment:
do we need to have an option to disable auto-heal? it would make things simpler if not

Reply:
yeah, I guess we probably do not need. will add this to the issue to track

Reply:
https://github.com/numaproj/numaplane/issues/24